## Medidas de dispersão

In [1]:
# carrega dependencias
import numpy as np
import pandas as pd

In [2]:
# monta drive
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [3]:
# carrega os dados
enem_sp = pd.read_csv('/content/drive/MyDrive/UDEMY - Estatística para Análise de Dados/Moreno Marcos/microdados_enem_2019_sp_tratado.csv', sep=',', encoding='iso-8859-1')
enem_sp

NO_MUNICIPIO_RESIDENCIA  IDADE SEXO  TP_ESTADO_CIVIL    RACA  \
0                 Santa Isabel     22    M                1   parda   
1                    São Paulo     17    F                1   parda   
2        São Bernardo do Campo     19    M                1   parda   
3                   Embu-Guaçu     17    F                1   parda   
4                    Guarulhos     23    F                1  branca   
...                        ...    ...  ...              ...     ...   
499935               São Paulo     18    F                1   parda   
499936               São Paulo     17    F                1   preta   
499937               São Paulo     17    F                1   parda   
499938               São Paulo     17    M                1   parda   
499939                  Araras     18    F                1  branca   

        TP_NACIONALIDADE NO_MUNICIPIO_NASCIMENTO SG_UF_NASCIMENTO  \
0                      1    Santa Izabel do Pará               PA   
1                      1                   Belém               PA   
2                      1   São Bernardo do Campo               SP   
3                      1             Itacoatiara               AM   
4                      1               São Paulo               SP   
...                  ...                     ...              ...   
499935                 1               São Paulo               SP   
499936                 1               São Paulo               SP   
499937                 1               São Paulo               SP   
499938                 1               São Paulo               SP   
499939                 1                  Araras               SP   

        TP_ST_CONCLUSAO  TP_ANO_CONCLUIU  ... NOTA_MT  TP_LINGUA  \
0                     1                4  ...   600.2     Inglês   
1                     2                0  ...   731.8     Inglês   
2                     2                0  ...   393.2   Espanhol   
3                     2                0  ...   419.2     Inglês   
4                     1                5  ...   796.1     Inglês   
...                 ...              ...  ...     ...        ...   
499935                1                1  ...   479.1   Espanhol   
499936                2                0  ...    43.2   Espanhol   
499937                2                0  ...   487.3     Inglês   
499938                2                0  ...   401.1     Inglês   
499939                2                0  ...   559.5     Inglês   

        TP_STATUS_REDACAO  COMP1  COMP2  COMP3  COMP4  COMP5  NOTA_REDACAO  \
0                     1.0  160.0  200.0  180.0  200.0  200.0         940.0   
1                     1.0  160.0  120.0  200.0  200.0  200.0         880.0   
2                     1.0   80.0  160.0  120.0  100.0  100.0         560.0   
3                     1.0  120.0  120.0  120.0  120.0   80.0         560.0   
4                     1.0  160.0  200.0  180.0  180.0  160.0         880.0   
...                   ...    ...    ...    ...    ...    ...           ...   
499935                1.0  100.0  120.0  120.0  140.0  100.0         580.0   
499936                1.0   80.0  120.0  140.0  120.0  100.0         560.0   
499937                1.0  120.0   80.0   60.0  120.0    0.0         380.0   
499938                1.0  160.0  120.0  160.0  140.0  160.0         740.0   
499939                1.0  120.0  120.0  120.0  120.0  140.0         620.0   

        INTERNET  
0            Sim  
1            Não  
2            Sim  
3            Sim  
4            Sim  
...          ...  
499935       Sim  
499936       Sim  
499937       Sim  
499938       Não  
499939       Sim  

[499940 rows x 35 columns]

Selecionamos a nota da matemáticas dos candidatos do ENEM em SP para calcularmos a sua variância (amostral).

In [4]:
enem_sp['NOTA_MT'].var()

33659.47050441148

In [5]:
enem_sp.NOTA_MT.var()

33659.47050441148

Temos aprox. 33,7k (pontos?) de dispersão ou variância das notas de matemática.

Calculamos agora o seu Desvio Padrão (amostral).

In [7]:
enem_sp.NOTA_MT.std()

183.46517518158993

Calculamos a média das notas de matématica.

In [9]:
enem_sp.NOTA_MT.mean()

497.5942403088371

Temos o desvio padrão de aprox. 183 (pontos para cima ou para baixo) de desvio das notas de matemática em relação à sua média.

Calculamos a desviasão absoluta média. Que é a média de cada nota substraída com a média das notas.

In [11]:
enem_sp.NOTA_MT.mad()

132.98102283464956

Calculamos a amplitude das notas

In [14]:
enem_sp.NOTA_MT.max() - enem_sp.NOTA_MT.min()

984.7

Agora fazemos o mesmo para a nota de redação

In [15]:
enem_sp.NOTA_REDACAO.var()

29082.774122780305

In [16]:
enem_sp.NOTA_REDACAO.std()

170.53672367786447

In [18]:
enem_sp.NOTA_REDACAO.mean()

593.0183221986639

Fazendo uma breve comparação temos uma dispersão menor nas notas de redação do que nas de matemática em relação à sua média. Em outras palavras, as notas de matemáticas estão mais "dispersas" do que as de redação.

Podemos visualizar um resumo geral descritivo de todas as variáveis associadas à nosssa tabela.

In [19]:
enem_sp.describe()

IDADE  TP_ESTADO_CIVIL  TP_NACIONALIDADE  TP_ST_CONCLUSAO  \
count  499940.000000    499940.000000     499940.000000    499940.000000   
mean       20.593387         1.053206          1.029902         1.476085   
std         6.465291         0.380866          0.230986         0.510351   
min        12.000000         0.000000          0.000000         1.000000   
25%        17.000000         1.000000          1.000000         1.000000   
50%        18.000000         1.000000          1.000000         1.000000   
75%        21.000000         1.000000          1.000000         2.000000   
max        91.000000         4.000000          4.000000         4.000000   

       TP_ANO_CONCLUIU      TP_ENSINO     CO_ESCOLA  TP_DEPENDENCIA_ADM_ESC  \
count    499940.000000  328350.000000  1.768940e+05           176894.000000   
mean          2.215758       1.002768  3.521138e+07                2.461327   
std           3.561417       0.052543  4.430392e+05                0.850072   
min           0.000000       1.000000  1.100306e+07                1.000000   
25%           0.000000       1.000000  3.501625e+07                2.000000   
50%           1.000000       1.000000  3.504496e+07                2.000000   
75%           3.000000       1.000000  3.529070e+07                2.000000   
max          13.000000       2.000000  5.300104e+07                4.000000   

       TP_LOCALIZACAO_ESC  TP_SIT_FUNC_ESC  ...        NOTA_CH        NOTA_LC  \
count       176894.000000    176894.000000  ...  499940.000000  499940.000000   
mean             1.012561         1.005014  ...     482.228054     491.941262   
std              0.111371         0.112845  ...     161.122963     156.247134   
min              1.000000         1.000000  ...       0.000000       0.000000   
25%              1.000000         1.000000  ...     450.600000     490.100000   
50%              1.000000         1.000000  ...     523.700000     538.300000   
75%              1.000000         1.000000  ...     578.100000     576.200000   
max              2.000000         4.000000  ...     822.700000     772.300000   

             NOTA_MT  TP_STATUS_REDACAO          COMP1          COMP2  \
count  499940.000000      499940.000000  499940.000000  499940.000000   
mean      497.594240           1.072105     125.000760     127.066808   
std       183.465175           0.584314      28.629513      39.846083   
min         0.000000           1.000000       0.000000       0.000000   
25%       427.900000           1.000000     120.000000     120.000000   
50%       511.800000           1.000000     120.000000     120.000000   
75%       617.900000           1.000000     140.000000     140.000000   
max       984.700000           9.000000     200.000000     200.000000   

               COMP3          COMP4          COMP5   NOTA_REDACAO  
count  499940.000000  499940.000000  499940.000000  499940.000000  
mean      118.760691     124.989359      97.200704     593.018322  
std        36.778882      33.879299      57.787427     170.536724  
min         0.000000       0.000000       0.000000       0.000000  
25%       100.000000     120.000000      60.000000     500.000000  
50%       120.000000     120.000000     100.000000     600.000000  
75%       140.000000     140.000000     140.000000     680.000000  
max       200.000000     200.000000     200.000000    1000.000000  

[8 rows x 25 columns]

DESCRITIVO:
Olhando para a tabela de acima:
- A notas de Linguagem e Códigos possui uma menor variabilidade (em torno à sua média: 492) do que as notas de Ciêncas Humanas e Matemática;
- Entre as notas das componentes, a COMP1 tem uma menor dispersão (em relação à sua média: 125) quando comparadas às outras componentes.

DIAGNOSTICO:
- Provavelente as provas de CH e MT são mais complexas do que a LC;
- Talvez, a prova da COMP1 é mais simples de resolver do que as outras componentes.

PRESCRIPTIVO:
- O INEP poderia recomendar às escolas de ensino médio reforçar as aulas de matemática e CH;
- Da mesma forma, para as outras componentes da redação.